common/common.cc

```diff
@@ -275,6 +275,9 @@ void appendFilesInDir(string_view basePath, const string &path, const sorbet::Un
             }
             appendFilesInDir(basePath, fullPath, extensions, recursive, result, absoluteIgnorePatterns,
                              relativeIgnorePatterns);
+        } else if (recursive && entry->d_type == DT_LNK && sorbet::FileOps::dirExists(fullPath)) {
```

Comment:
For symlinks, would we want to avoid cycles? Keep a set of resolved symlinks (abs path) somewhere and don't revisit?

Reply:
@alexchow I noted my personal view on this in the description but I _feel_ like symlink cycles support is not necessary (it may not be worth the effort but the `shared_ptr` and default argument may make the code quality impact bearable).

I have pushed a change to do the check needed (still need to test locally) so you can evaluate if it's worth it